consensus/src/quorum_store/batch_aggregator.rs

```diff
@@ -133,7 +136,7 @@ impl BatchAggregator {
     fn is_missed_fragment(&self, batch_id: BatchId, fragment_id: usize) -> bool {
         match self.batch_id {
             Some(self_batch_id) => {
-                if batch_id > self_batch_id {
+                if self_batch_id.nonce != batch_id.nonce || batch_id > self_batch_id {
```

Comment:
Again, should not we store the previous nonce and compare to prevent reply attacks? What if in the future we stop using the noise protocol?